### DISCUSSION TITLE
How we can enable github from firewall

### DISCUSSION THREAD

**Discussion by nikhilinnobit** [community/community#44738] [New to GitHub]
what ports and IP needs to be enable from firewall to give access to github only ?

**Comment by jge162** [comment]
Hi @nikhilinnobit here is the links you need and To access GitHub from behind a firewall, you will need to allow traffic to the following ports and IP addresses:

Ports:

TCP port 22 for SSH
TCP port 80 and 443 for HTTPS
IP addresses:

GitHub's IP ranges for web and git traffic: https://help.github.com/en/github/authenticating-to-github/about-githubs-ip-addresses
GitHub's IP ranges for SSH git access: https://help.github.com/en/github/authenticating-to-github/connecting-to-github-with-ssh

**Reply by jantman** [reply]
Looks like the SSH IPs are no longer listed there... though I also went back through the history for that repo and couldn't find them there previously either.

**Comment by nikhilinnobit** [comment]
Hi @jge162 

thanks for the reply, but i tried same on aws security group, that i allowed 80, 443 and 9418 port for github IP from security group outbound rules but still i m not able to clone the code

**Reply by jge162** [reply]
Hey Niki, could you provide more information about your set up and the exact error message you are receiving.

**Reply by jge162** [reply]
> thanks for the reply, but i tried same on aws security group, that i allowed 80, 443 and 9418 port for github IP from security group outbound rules but still i m not able to clone the code

@nikhilinnobit please note that GitHub primarily uses port 22 (SSH) and port 9418 (git) for communication. To allow access to GitHub, NIki you will need to open port 22 and 9418 on your firewall, and allow traffic to GitHub's IP addresses. However, it is worth noting that GitHub utilizes a large range of IP addresses, so it may be more secure to allow access to all IPs instead of specifying specific IPs. Additionally, you should also ensure that the security group inbound rules for port 22 and 9418 is open.

**Reply by jge162** [reply]
@nikhilinnobit did that work for you?

**Reply by jge162** [reply]
> Hi @jge162
> 
> thanks for the reply, but i tried same on aws security group, that i allowed 80, 443 and 9418 port for github IP from security group outbound rules but still i m not able to clone the code

Hopefully it worked for you. @nikhilinnobit

**Comment by 30afloresrojas** [comment]
thank you for keeping the fire wall standing

**Comment by Kulagin-G** [comment]
About IP ranges for fw whitelist - you can gain listed networks from https://api.github.com/meta
